File: yunpian/python/useAPI.py
#!/usr/local/bin/python
#-*-coding:utf-8-*-

#author: jacky
# Time: 14-2-22 下午11:48uthor: jacky
# Desc: 短信http接口的python代码调用示例
# https://www.yunpian.com/api/demo.html
import httplib
import urllib
import json
#服务地址
host = "yunpian.com"
#端口号
port = 80
#版本号
version = "v1"
#查账户信息的URI
user_get_uri = "/" + version + "/user/get.json"
#智能匹配模版短信接口的URI
sms_send_uri = "/" + version + "/sms/send.json"
#模板短信接口的URI
sms_tpl_send_uri = "/" + version + "/sms/tpl_send.json"
#语音短信接口的URI
sms_voice_send_uri = "/" + version + "/voice/send.json"
#语音验证码
voiceCode = 1234
def get_user_info(apikey):
    """
    取账户信息
    """
    conn = httplib.HTTPConnection(host , port=port)
    headers = {"Content-type": "application/x-www-form-urlencoded", "Accept": "text/plain"}
    conn.request('POST',user_get_uri,str( 'apikey=' + apikey))
    response = conn.getresponse()
    response_str = response.read()
    conn.close()
    return response_str

def send_sms(apikey, text, mobile):
    """
    能用接口发短信
    """
    params = urllib.urlencode({'apikey': apikey, 'text': text, 'mobile':mobile})
    headers = {"Content-type": "application/x-www-form-urlencoded", "Accept": "text/plain"}
    conn = httplib.HTTPConnection(host, port=port, timeout=30)
    conn.request("POST", sms_send_uri, params, headers)
    response = conn.getresponse()
    response_str = response.read()
    conn.close()
    return response_str

def tpl_send_sms(apikey, tpl_id, tpl_value, mobile):
    """
    模板接口发短信
    """
    params = urllib.urlencode({'apikey': apikey, 'tpl_id':tpl_id, 'tpl_value': tpl_value, 'mobile':mobile})
    headers = {"Content-type": "application/x-www-form-urlencoded", "Accept": "text/plain"}
    conn = httplib.HTTPConnection(host, port=port, timeout=30)
    conn.request("POST", sms_tpl_send_uri, params, headers)
    response = conn.getresponse()
    response_str = response.read()
    conn.close()
    return response_str

def send_voice_sms(apikey, code, mobile):
    """
    能用接口发短信
    """
    params = urllib.urlencode({'apikey': apikey, 'code': code, 'mobile':mobile})
    headers = {"Content-type": "application/x-www-form-urlencoded", "Accept": "text/plain"}
    conn = httplib.HTTPConnection(host, port=port, timeout=30)
    conn.request("POST", sms_voice_send_uri, params, headers)
    response = conn.getresponse()
    response_str = response.read()
    conn.close()
    return response_str

if __name__ == '__main__':
    apikey = "a4b3f38430ef7be6704b03181e76e7ca"
    mobile = "18812412312"
    text = "您的验证码是1234【云片网】"
    #查账户信息
    print(get_user_info(apikey))
    #调用智能匹配模版接口发短信
    print send_sms(apikey,text,mobile)
    #调用模板接口发短信
    tpl_id = 1 #对应的模板内容为：您的验证码是#code#【#company#】
    tpl_value = '#code#=1234&#company#=云片网'
    print(tpl_send_sms(apikey, tpl_id, tpl_value, mobile))
    #调用模板接口发语音短信
    print (send_voice_sms(apikey,voiceCode,mobile))

File: yunpian/nodejs/useAPI.js
// 修改为您的apikey.可在官网（http://www.yuanpian.com)登录后用户中心首页看到
var apikey = 'a4b3f38430ef7be6704b03181e76e7ca';
// 修改为您要发送的手机号码，多个号码用逗号隔开
var mobile = '18912312312';
// 修改为您要发送的短信内容
var text = '【云片网】您的验证码是1234';
// 指定发送的模板编号
var tpl_id = 1;
// 指定发送模板的内容
var tpl_value =  '#code#=1234&#company#=yunpian';
// 语音短信的内容
var code = 1234;
// 查询账户信息HTTP地址
var get_user_info_uri = '/v1/user/get.json';
// 智能匹配模板发送HTTP地址
send_sms_uri = '/v1/sms/send.json';
// 指定模板发送接口HTTP地址
send_tpl_sms_uri = '/v1/sms/tpl_send.json';
// 发送语音验证码接口HTTP地址
send_voice_uri = '/v1/voice/send.json';



var http = require('http');  
  
var qs = require('querystring');  

query_user_info(get_user_info_uri,apikey);

send_sms(send_sms_uri,apikey,mobile,text);

send_tpl_sms(send_tpl_sms_uri,apikey,mobile,tpl_id,tpl_value);

send_voice_sms(send_voice_uri,mobile,code);
function query_user_info(uri,apikey){
    var post_data = {  
    'apikey': apikey,  
    };//这是需要提交的数据
    var content = qs.stringify(post_data);  
    post(uri,content);
}

function send_sms(uri,apikey,mobile,text){
    var post_data = {  
    'apikey': apikey,  
    'mobile':mobile,
    'text':text,
    };//这是需要提交的数据  
    var content = qs.stringify(post_data);  
    post(uri,content);
}

function send_tpl_sms(uri,apikey,mobile,tpl_id,tpl_value){
    var post_data = {  
    'apikey': apikey,
    'mobile':mobile,
    'tpl_id':tpl_id,
    'tpl_value':tpl_value,  
    };//这是需要提交的数据  
    var content = qs.stringify(post_data);  
    post(uri,content); 
}
function send_voice_sms(uri,apikey,mobile,code){
    var post_data = {  
    'apikey': apikey,
    'mobile':mobile,
    'code':code,
    };//这是需要提交的数据  
    var content = qs.stringify(post_data);  
    post(uri,content); 
}
function post(uri,content){
    var options = {  
        hostname: 'www.yunpian.com',  
        port: 80,  
        path: uri,  
        method: 'POST',  
        headers: {  
            'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'  
        }  
    };
    var req = http.request(options, function (res) {  
        // console.log('STATUS: ' + res.statusCode);  
        // console.log('HEADERS: ' + JSON.stringify(res.headers));  
        res.setEncoding('utf8');  
        res.on('data', function (chunk) {  
            console.log('BODY: ' + chunk);  
        });  
    }); 
    req.write(content);  
  
    req.end();   
}

